src/shared/libcrypt-util.c

```diff
@@ -31,7 +31,7 @@
 
 int make_salt(char **ret) {
 
-#if HAVE_CRYPT_GENSALT_RA
+#if ((HAVE_CRYPT_GENSALT_RA) && (HAVE_CRYPT_PREFERRED_METHOD))
```

Comment:
hmm, i think a nicer fix would be to just hardcode `$6$` as default prefix if crypt_preferred_method is not available. i.e. see the other ifdef branch here, it also uses `$6$` as default prefix. i.e. instead of bypassing the whole crypt_gensalt_ra() branch just add a fallback for the crypt_preferred_method() call